src/encoder.c

```diff
@@ -466,17 +466,25 @@ const OSSL_DISPATCH p11prov_rsa_encoder_spki_pem_functions[] = {
 /* ECDSA */
 
 struct ecdsa_key_point {
-    ASN1_OBJECT *curve;
+    union {
+        ASN1_OBJECT *object;
+        ASN1_STRING *sequence;
+        void *p;
```

Comment:
why do you need p?

you can always just set k->curve.object = NULL when you need to clear this.